src/presets/aws-amplify.ts

```diff
@@ -0,0 +1,126 @@
+import { fileURLToPath } from "node:url";
+import { resolve } from "node:path";
+import { writeFile } from "node:fs/promises";
+import { joinURL } from "ufo";
+import { defineNitroPreset } from "../preset";
+import type { Nitro } from "../types";
+import {
+  AmplifyDeployManifest,
+  AmplifyRoute,
+  AmplifyRouteTarget,
+} from "../types/presets/aws-amplify";
+
+export const awsAmplify = defineNitroPreset({
+  extends: "node-server",
+  entry: fileURLToPath(new URL("entry.ts", import.meta.url)),
+  output: {
+    dir: "{{ rootDir }}/.amplify-hosting",
+    serverDir: "{{ output.dir }}/compute/default",
+    publicDir: "{{ output.dir }}/static{{ baseURL }}",
+  },
+  commands: {
+    preview: "node ./compute/default/server.js",
+  },
+  hooks: {
+    async compiled(nitro) {
+      await writeAmplifyFiles(nitro);
+    },
+  },
+});
+
+async function writeAmplifyFiles(nitro: Nitro) {
+  const outDir = nitro.options.output.dir;
+
+  // Generate routes
+  const routes: AmplifyRoute[] = [];
+
+  let hasWildcardPublicAsset = false;
+
+  if (nitro.options.awsAmplify?.imageOptimization) {
+    const { path, cacheControl } = nitro.options.awsAmplify?.imageOptimization;
+    routes.push({
+      path,
+      target: {
+        kind: "ImageOptimization",
+        cacheControl,
+      },
+    });
+  }
+
+  const computeTarget = {
+    kind: "Compute",
+    src: "default",
+  } as AmplifyRouteTarget;
+
+  for (const publicAsset of nitro.options.publicAssets) {
+    if (!publicAsset.baseURL || publicAsset.baseURL === "/") {
+      hasWildcardPublicAsset = true;
+      continue;
+    }
+    routes.push({
+      path: `${publicAsset.baseURL!.replace(/\/$/, "")}/*`,
+      target: {
+        cacheControl:
+          publicAsset.maxAge > 0
+            ? `public, max-age=${publicAsset.maxAge}, immutable`
+            : undefined,
+        kind: "Static",
+      },
+      fallback: publicAsset.fallthrough ? computeTarget : undefined,
+    });
+  }
+  if (hasWildcardPublicAsset) {
+    routes.push({
+      path: "/*.*",
+      target: {
+        kind: "Static",
+      },
+      fallback: computeTarget,
+    });
+  }
+  routes.push({
+    path: "/*",
+    target: computeTarget,
+    fallback:
+      hasWildcardPublicAsset && nitro.options.awsAmplify.computeStaticFallback
+        ? {
+            kind: "Static",
+          }
+        : undefined,
+  });
+
+  // Prefix with baseURL
+  for (const route of routes) {
+    if (route.path !== "/*") {
+      route.path = joinURL(nitro.options.baseURL, route.path);
+    }
+  }
+
+  // Generate deploy-manifest.json
+  const deployManifest: AmplifyDeployManifest = {
+    version: 1,
+    routes,
+    imageSettings: nitro.options.awsAmplify?.imageSettings || undefined,
+    computeResources: [
+      {
+        name: "default",
+        entrypoint: "server.js",
+        runtime: "nodejs18.x",
+      },
+    ],
+    framework: {
+      name: nitro.options.framework.name || "nitro",
```

Comment:
If I don't provide `framework` in `nuxt.config.ts`, I get an error:
```
 ERROR  Cannot read properties of undefined (reading 'name')                                                                                                   
  at writeAmplifyFiles (amplify/nitro.config.ts:112:37)
```

So is `framework` going to be a required field in `nuxt.config.ts` ?

Reply:
I guess you are testing nitro without this PR. It has default so that shouldn't happen.